debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java

```diff
@@ -281,6 +282,33 @@ public void execute(String desc, Consumer<MongoClient> operation) {
             }
         }
 
+        /**
+         * Execute the supplied operation using the primary, blocking until a primary is available. Whenever the operation stops
+         * (e.g., if the primary is no longer primary), then restart the operation using the current primary.
+         *
+         * @param desc the description of the operation, for logging purposes
+         * @param operation the operation to be performed on the primary
+         * @return return value of the executed operation
+         */
+        public <T> T execute(String desc, Function<MongoClient, T> operation) {
+            final Metronome errorMetronome = Metronome.sleeper(PAUSE_AFTER_ERROR, Clock.SYSTEM);
+            while (true) {
```

Comment:
Will this loop terminate when the connector is stopped?

Reply:
Good point; indeed it seems they'd keep running if the connector is shut down and they don't complete normally. I'll log an issue.